File: SoftUni Lessons/Python Development/Python Basics April 2019/Lessons and Problems/09.1 - While Loop More Exercises/01. Dishwasher.py
count_bottles = int(input())
volume = count_bottles * 750

count_dishes = 0
count_pots = 0

command = None
counter = 0

while True:
    if volume < 0:
        break
    command = input()
    if command == "End":
        break
    if volume - int(command) < 0:
        break
    counter += 1
    if counter == 3:
        volume -= int(command) * 15
        count_pots += int(command)
        counter = 0
    else:
        volume -= int(command) * 5
        count_dishes += int(command)

if command == "End" and volume >= 0:
    print(f"Detergent was enough!")
    print(f"{count_dishes} dishes and {count_pots} pots were washed.")
    print(f"Leftover detergent {volume} ml.")
else:
    print(f"Not enough detergent, {abs(volume)} ml. more necessary!")

File: SoftUni Lessons/JavaScript Development/JavaScript Basics March 2020/14. УПРАЖНЕНИЕ ЦИКЛИ - ЧАСТ 2/While-Loop - More Exercises/01. Dishwasher.js
// Къс линк: https://git.io/JvjRq

function dishwasher(params) {
    let detergent = 750 * Number(params.shift());
    let dishes = 0;
    let pots = 0

    let timesInHell = 0;
    while('You wash them dirty bit**s') {
        let plates = params.shift();
        if (plates == 'End') {break;}
        
        timesInHell += 1
        plates = Number(plates);
        if (timesInHell % 3 == 0) {detergent -= (plates * 15); pots += plates;}
        else {detergent -=plates * 5; dishes += plates}
        
        if (detergent < 0) {break;}
    }

    if (detergent >= 0) {
        console.log(`Detergent was enough!`);
        console.log(`${dishes} dishes and ${pots} pots were washed.`);
        console.log(`Leftover detergent ${detergent} ml.`);
    } else {
        console.log(`Not enough detergent, ${Math.abs(detergent)} ml. more necessary!`);
    }
}

dishwasher([2, 53, 65, 55, 'End']);  // Expected Output:
// Detergent was enough!
// 118 dishes and 55 pots were washed.
// Leftover detergent 85 ml.


dishwasher([1, 10, 15, 10, 12, 13, 30]);  // Expected Output:
// Not enough detergent, 100 ml. more necessary!
